modin/numpy/arr.py

```diff
@@ -249,6 +289,31 @@ def _update_inplace(self, new_query_compiler):
             sib._query_compiler = new_query_compiler
         old_query_compiler.free()
 
+    def _validate_axis(self, axis):
+        """
+        Check that the provided axis argument is valid on this array.
+
+        Raises numpy.AxisError if the axis is invalid.
+
+        Parameters
+        ----------
+        axis : int, optional
+            The axis argument passed to the function.
+
+        Returns
+        -------
+        int, optional
+            Axis to apply the function over (None, 0, or 1).
+        """
+        if axis is not None and axis < 0:
```

Comment:
In which cases `axis` can be < 0?

Reply:
Many numpy functions (I have not exhaustively tested which ones in particular) can take a negative axis argument, which is then treated as essentially `_ndim + axis` (for example, `axis=-1` in a 2D array would apply the function as if `axis=1`).

```python
>>> import numpy as np
>>> np.array([[1,2],[3,4]])
array([[1, 2],
       [3, 4]])
>>> np.array([[1,2],[3,4]]).sum(axis=-1)
array([3, 7])
```

Reply:
got it, thanks